File: lc_soln/1347_minimum_number_of_steps_to_make_two_strings_anagram/solution.cpp
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
    int minSteps(string s, string t) {
        vector<int> counter_s(26, 0);
        vector<int> counter_t(26, 0);
        for (int i = 0; i < s.size(); ++i) {
            counter_s[s[i] - 'a']++;
            counter_t[t[i] - 'a']++;
        }

        int result = 0;
        for (int i = 0; i < 26; ++i) {
            if (int diff = counter_s[i] - counter_t[i]; diff > 0) {
                result += diff;
            }
        }
        return result;
    }
};

File: lc_soln/1347_minimum_number_of_steps_to_make_two_strings_anagram/solution.py
class Solution:
    def minSteps(self, s: str, t: str) -> int:
        if s == t:
            return 0

        counter_s = Counter(s)
        counter_t = Counter(t)
        result = 0
        for c, v in counter_s.items():
            if c not in counter_t:
                result += v
            elif (diff := v - counter_t[c]) > 0:
                result += diff
        return result
